report/thesis.tex

```diff
@@ -34,20 +34,37 @@
 \chapter{Introduction}
 %%%%%%%%%%%%%%%%%%%%%%
 
-TODO
+Tendermint is a partially synchronous protocol which provides a simple and efficient algorithm to solve the problem of Byzantine fault tolerant (BFT) consensus in a distributed network of processes. As every other BFT consensus protocol, the Tendermint algorithm relies on the assumption that byzantine processes are limited to a certain threshold: this assumption allows correct processes to have enough voting power to approve new blocks and guarantees the correctness and validity of the blockchain throughout the entire execution of the protocol. 
+
+When this assumption cannot be ensured, malicious processes can cause the failure of the algorithm and create deviations from the main chain (better known as forks). In such cases, a valid protocol should provide some reliable guarantees that, under certain constraints, allow the protocol to recover from the fault, detect the faulty processes responsible of a fork and eventually re-store the conditions for the safe execution of the consensus algorithm.     
```

Comment:
Not sure what you mean by "valid" in this sentence.

```suggestion
When this assumption cannot be ensured, malicious processes can cause the failure of the algorithm and create deviations from the main chain (better known as forks). In such cases, a failover solution should provide some reliable guarantees that, under certain constraints, allow the consensus algorithm to recover from the fault, detect the faulty processes responsible of a fork and eventually re-store the conditions for the safe execution of the consensus algorithm.     
```

Reply:
Yeah "failover" is more appropriate in this case, valid was way too general. Fixed